src/coprocessor/dag/aggr_fn/impl_first.rs

```diff
@@ -0,0 +1,241 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::marker::PhantomData;
+
+use cop_codegen::AggrFunction;
+use cop_datatype::EvalType;
+use tipb::expression::{Expr, ExprType, FieldType};
+
+use crate::coprocessor::codec::data_type::*;
+use crate::coprocessor::codec::mysql::Tz;
+use crate::coprocessor::dag::expr::EvalContext;
+use crate::coprocessor::dag::rpn_expr::{RpnExpression, RpnExpressionBuilder};
+use crate::coprocessor::Result;
+
+/// The parser for FIRST aggregate function.
+pub struct AggrFnDefinitionParserFirst;
+
+impl super::AggrDefinitionParser for AggrFnDefinitionParserFirst {
+    fn check_supported(&self, aggr_def: &Expr) -> Result<()> {
+        assert_eq!(aggr_def.get_tp(), ExprType::First);
+        super::util::check_aggr_exp_supported_one_child(aggr_def)
+    }
+
+    fn parse(
+        &self,
+        mut aggr_def: Expr,
+        time_zone: &Tz,
+        src_schema: &[FieldType],
+        out_schema: &mut Vec<FieldType>,
+        out_exp: &mut Vec<RpnExpression>,
+    ) -> Result<Box<dyn super::AggrFunction>> {
+        use cop_datatype::FieldTypeAccessor;
+        use std::convert::TryFrom;
+        assert_eq!(aggr_def.get_tp(), ExprType::First);
+        let child = aggr_def.take_children().into_iter().next().unwrap();
+        let eval_type = EvalType::try_from(child.get_field_type().tp()).unwrap();
+
+        // FIRST outputs one column with the same type as its child
+        out_schema.push(aggr_def.take_field_type());
+
+        // FIRST doesn't need to cast, so using the expression directly.
```

Comment:
Feel weird to add the comment here. 🤔

Reply:
Yes, seems too trivial... Let me delete it.